### PR TITLE
Fix the MCP Client ConfigZ page

### PR DESCRIPTION
**Pull request by ozevren**
Fixes https://github.com/istio/istio/issues/12297

### CODE REVIEW

**Comment by ozevren** [comment]
Port of https://github.com/istio/istio/pull/12302

File: pkg/mcp/configz/configz_test.go [hunk 1]
```diff
@@ -131,28 +129,14 @@ func testConfigZWithNoRequest(t *testing.T, baseURL string) {
...
-			time.Sleep(time.Millisecond * 100)
-			continue
-		}
-		return
-	}
-	t.Fatal("Both collections, the initial request, and a recent ACK request should have been displayed")
-}
-
-func testConfigJWithOneRequest(t *testing.T, baseURL string) {
+func testConfigJWithTwoRequests(t *testing.T, baseURL string) {
```
Comment (therealmitchconnors) [inline pkg/mcp/configz/configz_test.go]: This looks like only one request is being used in the test.  Am I missing something?

Comment (ozevren) [reply pkg/mcp/configz/configz_test.go]: It is about having two "LatestRequests" in the list. I've updated the test name to clarify.
The  LatestRequest check is not visible in the diff. You can see it by expanding the diff output.

**Pull request merged**